### PR TITLE
Move bin/jekyll to exe/jekyll to prevent collision with binstubs

### PR DESCRIPTION
**Pull request by parkr**
If you run `bundle install --binstubs` in this repo, lots of executables are created for the dependencies of this project. The directory is not ignored because we have historically kept our executable (i.e. the `jekyll` CLI command) in this directory.

This PR moves it to `exe/jekyll` and ignores the `bin/` directory.

### CODE REVIEW

**Comment by SuriyaaKudoIsc** [comment]
:+1:

**Comment by envygeeks** [comment]
:shipit:

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,7 @@
 /vendor
 Gemfile.lock
 _site/
+bin/
```
Comment (envygeeks) [inline .gitignore]: You should probably add !bin/jekyll because we need that to be tracked.

Comment (parkr) [inline .gitignore]: The PR body explains why that is not necessary.

**Comment by parkr** [comment]
> Looks like some of our stuff is tightly coupled to bin?

Yep, in `features`. Fixing.

File: .gitignore (hunk 1 above)
Comment (envygeeks) [inline .gitignore]: @parkr yeah I noticed that after I commented :blush:

**Comment by parkr** [comment]
@jekyllbot: merge +minor

**Pull request merged**